### PR TITLE
Add Android chat demo

### PR DESCRIPTION
**Pull request by ziyiz-amzn**
**Description of changes:**
Add an Android demo to demonstrate how to quickly add chat feature into iOS app using Amazon Chime SDK for messaging.

**Testing**

1. How did you test these changes? manually
2. Can these changes be tested using one of the demo application? If yes, which demo application can be used to test it? N/A

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by zhinang-amazon — APPROVED** [review]

File: apps/chat-android/.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.gradle/
+.idea/
+app/build/
+local.properties
+messaging-websocket/build/
```
Comment (linsang21) [inline apps/chat-android/.gitignore:5]: Doesn't `build/` cover this?

File: apps/chat-android/.gitignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+build/
```
Comment (linsang21) [inline apps/chat-android/.gitignore:6]: Nit: add newline at end, same for other files

File: apps/chat-android/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Amazon Chime SDK Android Chat Demo
+
+## Overview
+
+By completing the steps below, you will learn how to run the Amazon Chime SDK Android chat demo applications on your mobile device or on a simulator. This application enables users to send and receive messages/image attachments in chat channels in real time. This application is designed in a way that can be used as a starting point for your own application, or as a quick way to explore the features of the Amazon Chime SDK messaging. Please see the [Build chat applications in iOS and Android with Amazon Chime SDK messaging](https://aws.amazon.com/blogs/business-productivity/build-chat-applications-in-ios-and-android-with-amazon-chime-sdk-messaging) for more information.
+
+<p align="center">
+<img src="./media/awsChimeSDKMessagingDemo.png" alt="image" width="80%"/>
```
Comment (linsang21) [inline apps/chat-android/README.md:8]: Nit: Some texts in the image are blocked by arrows / lines, readability is a little compromised. Do you think we can adjust?

File: apps/chat-android/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+   ```
+2. Navigate to the root folder of the Android chat demo application.
+   ```
+   cd apps/chat-android
+   ```
+3. Open the project in Android Studio.
+4. Open `app/src/main/res/raw/amplifyconfiguration.json` and update the configuration with the values from outputs from the deployment of AWS CloudFormation template.
+5. Open `app/src/main/java/com/amazonaws/services/chime/sdkdemo/common/AppConstants.kt` and update `APP_INSTANCE_ARN` and `API_GATEWAY_INVOKE_URL`.
+   ```
+    const val APP_INSTANCE_ARN = "<appInstanceArn fromm CloudFormation Outputs>"
```
Comment (linsang21) [inline apps/chat-android/README.md:33]: typo in this line and the next line: from

File: apps/chat-android/README.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# Amazon Chime SDK Android Chat Demo
+
+## Overview
+
+By completing the steps below, you will learn how to run the Amazon Chime SDK Android chat demo applications on your mobile device or on a simulator. This application enables users to send and receive messages/image attachments in chat channels in real time. This application is designed in a way that can be used as a starting point for your own application, or as a quick way to explore the features of the Amazon Chime SDK messaging. Please see the [Build chat applications in iOS and Android with Amazon Chime SDK messaging](https://aws.amazon.com/blogs/business-productivity/build-chat-applications-in-ios-and-android-with-amazon-chime-sdk-messaging) for more information.
```
Comment (devalevenkatesh) [inline apps/chat-android/README.md:5]: Could you please fix the blog post broken link?

Also, The messaging SDK chat demo README calls Amazon Chime SDK for messaging and not Amazon Chime SDK messaging hence suggest to follow the same convention in this README as well as Android chat demo README. The blog post says Amazon Chime SDK messaging but the "for" IMO makes it a bit more clear.

File: apps/chat-android/.gitignore (hunk 1 above)
Comment (ziyiz-amzn) [reply apps/chat-android/.gitignore:5]: updated

File: apps/chat-android/README.md (hunk 3 above)
Comment (ziyiz-amzn) [reply apps/chat-android/README.md:8]: Updated.

File: apps/chat-android/README.md (hunk 4 above)
Comment (ziyiz-amzn) [reply apps/chat-android/README.md:33]: good catch

File: apps/chat-android/README.md (hunk 5 above)
Comment (ziyiz-amzn) [reply apps/chat-android/README.md:5]: Replied in iOS PR.

**Review by linsang21 — APPROVED** [review]

**Review by hokyungh — APPROVED** [review]

**Review by devalevenkatesh — APPROVED** [review]

**Pull request merged**